### ISSUE TITLE
ML Script

### ISSUE THREAD

**Issue by nicovandenhooff** [UBC-MDS/online-shoppers-purchasing-intention#10]
Nico to complete, goal of Wednesday completion.

**Comment by nicovandenhooff** [comment]
@ytz @arijc76 

A couple things to discuss tomorrow...

I just read Florencia's latest issue on GitHub she notes: **For this project, one kind of model is fine (you should tune the one though if there is a hyperparameter!)**

I'm thinking it still makes more sense to train multiple models so that we can perform a round of model selection.  However I think we may only want to pick one of XGBoost, CatBoost or LightGBM (in addition to all of the sklearn models) in order to try to keep this project a bit more clean.  I'm thinking XGBoost.

Further, if we have multiple models, I think we may need to split the ML script into two separate scripts as below:
 
1. The first script would create, tune, and return the cv scores and confusion matrices for our base models (I will be done this tonight).  This script would also include feature engineering if we decide to move it over based on the other Github issue.
2. We then analyze the output of the first script, and then write the second script to focus on hyperparameter tuning of the best performing model from the script above.  This script uses the best tuned model to return the final results on the test set.

Let's chat more on these tomorrow.

Cheers,
Nico
